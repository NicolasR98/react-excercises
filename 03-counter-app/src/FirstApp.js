import React from "react";
import PropTypes from "prop-types";

//Functional Components

const FirstApp = ({ greet, subtitle }) => {
  const str = greet;
  const obj = {
    name: "nicolas",
    age: 23,
  };

  return (
    <>
      <h1>{str}</h1>
      <p>{subtitle}</p>
      {/* <pre>{JSON.stringify(obj, null, 3)}</pre> */}
    </>
  );
};

//Defines the prop type for the props
//Like typescript
FirstApp.propTypes = {
  greet: PropTypes.string.isRequired,
};

//Declare default prop
FirstApp.defaultProps = {
    subtitle: `I'm a subtitle!`
}

export default FirstApp;
