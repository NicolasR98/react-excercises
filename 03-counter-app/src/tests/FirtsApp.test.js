import { render, shallow } from "enzyme";
import "@testing-library/jest-dom";
import React from "react";
import FirstApp from "../FirstApp";

describe("Test on <FirstApp />", () => {
  test("should display <FirstApp /> correctly", () => {
    const greet = "Hello world";
    const wrapper = shallow(<FirstApp greet={greet} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Should display subtitle correctly", () => {
    const subtitle = `I am a subtitle!`;
    const greet = "Hello world";
    const wrapper = shallow(<FirstApp greet={greet} subtitle={subtitle} />);

    const paragraphText = wrapper.find("p").text();

    expect(paragraphText).toBe(subtitle);
  });
});
