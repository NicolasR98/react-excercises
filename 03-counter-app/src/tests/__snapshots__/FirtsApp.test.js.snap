// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstApp /> should display <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hello world
  </h1>
  <p>
    I'm a subtitle!
  </p>
</Fragment>
`;
